Added tests for CCP searches

mpelzsherman opened the pull request:
https://github.com/department-of-veterans-affairs/va.gov-team/issues/10448

Enhanced test coverage for CCP searches.

mpelzsherman committed:
Added test for CCP searches

https://github.com/department-of-veterans-affairs/va.gov-team/issues/10448

brangi commented on src/applications/facility-locator/tests/e2e/search.cypress.spec.js:
```diff
@@ -83,9 +86,28 @@ describe('Facility search', () => {
+
+    cy.get('#facility-search-results').should('not.exist');
+  });
+
+  it('finds community dentists', () => {
+    cy.visit('/find-locations');
+
```
You should mock the correct data and verify

Do something like this:

```    cy.route(
      'GET',
      '/v1/facilities/cpp/',
      'fx:constants/<CCP_MOCK_DATA>',
    ).as('searchCCP');``

mpelzsherman replied:
I was able to use the `MockApi` instead by checking for `window.Cypress` in `src/applications/facility-locator/api/index.js`.

mpelzsherman committed:
Use MockApi for Cypress tests

mpelzsherman committed:
MOAR tests!

mpelzsherman committed:
Test improvements:

- Added map marker checks to search.cypress.spec.js
- Refactored LocationDistance, changed VaFacilityResult.unit.spec.jsx to use explicit find rather than snapshots

mpelzsherman committed:
update remaining search result tests

make heading level consistentg (h3)

added id's to facilitate testing

mpelzsherman committed:
Improve tests for LocationPhoneLink and LocationDirectionsLink

mpelzsherman commented on src/applications/facility-locator/api/index.js:
```diff
@@ -4,7 +4,7 @@ import environment from 'platform/utilities/environment';
 // To use vets-api data locally, replace the following with this:
 // export default LiveApi;
-export default (environment.isLocalhost() &&
+export default ((environment.isLocalhost() || window.Cypress) &&
```
This change makes it so we always use the MockApi for Cypress tests.

mpelzsherman committed:
Added test for LocationDistance

brangi reviewed (approved):
Looking good!

mpelzsherman committed:
update cypress test
